### DISCUSSION TITLE
How to auto-install plugins on the first `nvim` start without user confirmation (a.k.a. noninteractive behaviour)

### DISCUSSION THREAD

**Discussion by simon-liebehenschel** [NvChad/NvChad#1841] [Q&A]
Should this question be moved to https://github.com/folke/lazy.nvim?

I install all in a Dockerfile. The first time I start `nvim`, I need to press `Y` manually to continue and install plugins. Then I am asked to press `ENTER`.

Are there any config files or neovim commands that enable auto-installing without a user manual interaction? I mean do not ask to press `y` on a keyboard, and install all immediately.

I think my Dockerfile is not important for my question, but just in case I will put it here.
```Dockerfile
# Inspired by https://github.com/MashMB/nvim-ide/blob/master/nvim/Dockerfile

FROM debian:11

ENV DEBIAN_FRONTEND noninteractive
# Set image locale.
ENV LANG en_US.UTF-8
ENV LANGUAGE en_US:en
ENV TZ=Europe/Warsaw

RUN rm -f /etc/apt/apt.conf.d/docker-clean; echo 'Binary::apt::APT::Keep-Downloaded-Packages "true";' > /etc/apt/apt.conf.d/keep-cache
RUN \
    --mount=type=cache,target=/var/cache/apt,sharing=locked \
    --mount=type=cache,target=/var/lib/apt,sharing=locked \
    apt-get update \
    # NvChad dependencies: ripgrep, build-essential, neovim
    && apt-get install -y \
    curl \
    fzf \
    ripgrep \
    tree \
    git \
    xclip \
    # nodejs <-- A newer version will be installed further. \
    npm \
    tzdata \
    unzip \
    wget \
    build-essential \
    && curl -sL install-node.vercel.app/lts | bash -s -- --yes \
    && wget https://github.com/neovim/neovim/releases/download/stable/nvim-linux64.deb \
    && apt-get install ./nvim-linux64.deb \
    && rm ./nvim-linux64.deb \
    && rm -rf /var/lib/apt/lists/* \
    \
    # Install NvChad.
    && git clone https://github.com/NvChad/NvChad ~/.config/nvim --depth 1
```

**Comment by siduck** [comment]
@AIGeneratedUsername the "y" prompt is made by nvchad config and not lazy.nvim

**Reply by simon-liebehenschel** [reply]
I am very new to Vim, Lua, etc. Sorry for a dumb question.
So, technically, I can update the nvchad config, and then in my Dockerfile I will be able to execute once `nvim` which will auto-install all plugins? The idea is to have a ready to use nvchad after a container is built.

**Reply by siduck** [reply]
@AIGeneratedUsername i have used docker very little, but dont you run nvim after the docker container is ready?

**Reply by simon-liebehenschel** [reply]
**Let's forget about Docker.** 

The question is: "how to start  (how to update its config files) NvChad so a user will not need to type `y` on the first start".  
You may just point at which exactly config file I need to look.

**Reply by siduck** [reply]
@AIGeneratedUsername we could use an install script but i dont want to, I want the repo to be 100% lua and the installation process as simple as possible. One thing i could do is wait for 5 seconds for the prompt, if its empty then just auto-install everything
